src/server/game/Maps/Map.cpp

```diff
@@ -2441,24 +2441,11 @@ float Map::GetWaterOrGroundLevel(uint32 phasemask, float x, float y, float z, fl
     return VMAP_INVALID_HEIGHT_VALUE;
 }
 
-float Map::GetHeight(float x, float y, float z, bool checkVMap /*= true*/, float maxSearchDist /*= DEFAULT_HEIGHT_SEARCH*/) const
+float Map::GetHeight(float x, float y, float z, bool checkVMap /*= true*/, float /*maxSearchDist*/ /*= DEFAULT_HEIGHT_SEARCH*/) const
 {
     // find raw .map surface under Z coordinates
-    float mapHeight = VMAP_INVALID_HEIGHT_VALUE;
-    if (GridMap* gmap = const_cast<Map*>(this)->GetGrid(x, y))
-    {
-        float gridHeight = gmap->getHeight(x, y);
-        if (G3D::fuzzyGe(z, gridHeight - GROUND_HEIGHT_TOLERANCE))
-            mapHeight = gridHeight;
-    }
-
-    float vmapHeight = VMAP_INVALID_HEIGHT_VALUE;
-    if (checkVMap)
-    {
-        VMAP::IVMapManager* vmgr = VMAP::VMapFactory::createOrGetVMapManager();
-        if (vmgr->isHeightCalcEnabled())
-            vmapHeight = vmgr->getHeight(GetId(), x, y, z, maxSearchDist);
-    }
+    float const mapHeight = GetGridHeight(x, y);
+    float const vmapHeight = checkVMap ? GetVmapDataFloorZ(x, y, z) : VMAP_INVALID_HEIGHT_VALUE;
```

Comment:
Why replace vmgr->getHeight with vmgr->getAreaAndLiquidData?

Reply:
Was looking for a function that could return the floor without a max search distance, as the default one was not enough so it would return an invalid vmap height iirc

Need to check it again to be sure

Reply:
@Golrag jump into irc :)

Reply:
@Aokromes Any news on this? 👍

Reply:
@yvomsx test it yourself, and give feedback.

Reply:
@Aokromes Give me a little, will review and post results.